apps/airtable-cache/src/app.ts

```diff
@@ -0,0 +1,226 @@
+import { fastify, FastifyReply, FastifyRequest } from 'fastify';
+import { fastifyCors } from '@fastify/cors';
+import axios, { AxiosResponse } from 'axios';
+import { errorHandlerPlugin } from './lib/errorHandlerPlugin';
+import { sha256 } from './lib/sha256';
+import {
+  CacheValue, clearValues, clearValuesContainingBody, getValue, setValue,
+} from './lib/cache';
+
+export const getInstance = async () => {
+  const instance = fastify();
+
+  await instance.register(errorHandlerPlugin);
+  await instance.register(fastifyCors);
+
+  await instance.register(async (i) => {
+    i.get('/', async (req, reply) => {
+      reply.redirect('https://github.com/bluedotimpact/bluedot/tree/master/apps/airtable-cache#readme');
+    });
+
+    i.get('/v0/meta/bases/:baseId/tables', cachedGetHandler);
+    i.get('/v0/:baseId/:tableId', cachedGetHandler);
+    i.get('/v0/:baseId/:tableId/:recordId', cachedGetHandler);
+
+    // TODO: clear out base schema
+    i.post('/v0/meta/bases/:baseId/tables', cacheClearingHandler(['baseSchema']));
+    i.post('/v0/meta/bases/:baseId/tables/:tableId/fields/:fieldId', cacheClearingHandler(['baseSchema']));
+
+    // TODO: clear out base schema, clear out all table data, and all records in that table
+    i.post('/v0/meta/bases/:baseId/tables/:tableId/fields', cacheClearingHandler(['baseSchema', 'tableData', 'tableRecords']));
+
+    // TODO: clear out table data with mentioned records, and mentioned record
+    i.patch('/v0/:baseId/:tableId', cacheClearingHandler(['mentionedRecords']));
+    i.put('/v0/:baseId/:tableId', cacheClearingHandler(['mentionedRecords']));
+    i.delete('/v0/:baseId/:tableId', cacheClearingHandler(['mentionedRecords']));
+    i.patch('/v0/:baseId/:tableId/:recordId', cacheClearingHandler(['mentionedRecords']));
+    i.put('/v0/:baseId/:tableId/:recordId', cacheClearingHandler(['mentionedRecords']));
+    i.delete('/v0/:baseId/:tableId/:recordId', cacheClearingHandler(['mentionedRecords']));
+    i.post('/v0/:baseId/:recordId/:fieldId/uploadAttachment', cacheClearingHandler(['mentionedRecords']));
+
+    // TODO: clear out all table data
+    i.post('/v0/:baseId/:tableId', cacheClearingHandler(['tableData']));
+
+    i.route({
+      method: ['DELETE', 'GET', 'HEAD', 'PATCH', 'POST', 'PUT'],
+      url: '/*',
+      handler: async (req, reply) => {
+        const airtableResponse = await proxyToAirtable(req);
+        return airtableToFastifyReply(airtableResponse, reply);
+      },
+    });
+  });
+
+  return instance;
+};
+
+const cachedGetHandler = async (req: FastifyRequest, reply: FastifyReply) => {
```

Comment:
Maybe consider setting up webhooks on Airtable so that if the records change through other routes e.g. different API changes, or manual edits in Airtable, the cache isn't stale.

Or at least make it very easy to clear the cache - maybe a web UI or something.